client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/index.tsx

```diff
@@ -42,13 +46,25 @@ const Launchpad: Step = ( { navigation, flow }: LaunchpadProps ) => {
 	);
 
 	if ( ! isLoggedIn ) {
-		window.location.replace( `/home/${ siteSlug }` );
+		redirectToSiteHome( siteSlug );
 	}
 
 	if ( ! siteSlug || fetchingSiteError?.error ) {
 		window.location.replace( '/home' );
 	}
 
+	function redirectToSiteHome( siteSlug: string | null ): void {
+		recordTracksEvent( 'calypso_launchpad_redirect_to_home', { flow: flow } );
+		window.location.replace( `/home/${ siteSlug }` );
+	}
+
+	function areLaunchpadTasksCompleted( flow: string | null ): boolean {
+		if ( isNewsletterFlow( flow ) ) {
+			return isFirstPostPublished;
+		}
+		return isSiteLaunched;
+	}
+
```

Comment:
The real change here is the addition and use of this method. Note that for non-newsletter sites, I'm not actually checking if *launchpad tasks are complete*, but rather if site is launched. This captures a wider array of situations - it's theoretically possible to have a site end up in 'launched' state without all launchpad tasks being complete. But in any case, if a site launched (other than newsletter), then the current full screen launchpad should not be shown. 

We could also check actual task status by checking against `site?.options?.launchpad_checklist_tasks_statuses?.site_launched`.